pyanaconda/modules/network/firewall/firewall.py

```diff
@@ -206,6 +207,18 @@ def set_disabled_services(self, disabled_services):
         self.disabled_services_changed.emit()
         log.debug("Services that will be explicitly disabled on the firewall: %s", self._disabled_services)
 
+    def collect_requirements(self):
+        """Return installation requirements for this module.
+
+        :return: a list of requirements
+        """
+        requirements = []
+
+        if self.firewall_seen:
+            requirements.append(Requirement.for_package("firewalld", reason="firewall configuration requested"))
```

Comment:
The attribute `reason` is being discussed at https://github.com/rhinstaller/anaconda/pull/2073#discussion_r311408871.